hs_swat_modelinstance/models.py

```diff
@@ -0,0 +1,550 @@
+            model_methods.save()
+        else:
+            raise ObjectDoesNotExist("No ModelMethods element was found for the provided id:%s" % kwargs['id'])
+
+    @classmethod
+    def remove(cls, element_id):
+        raise ValidationError("ModelMethods element of a resource can't be deleted.")
+
+
+class SWATModelParameters(AbstractMetaDataElement):
+    term = 'SWATModelParameters'
+    has_crop_rotation = models.BooleanField(default=False)
+    has_title_drainage = models.BooleanField(default=False)
+    has_point_source = models.BooleanField(default=False)
+    has_fertilizer = models.BooleanField(default=False)
+    has_tillage_operation = models.BooleanField(default=False)
+    has_inlet_of_draining_watershed = models.BooleanField(default=False)
+    has_irrigation_operation = models.BooleanField(default=False)
+    has_other_parameters = models.CharField(max_length=500, null=True, blank=True)
+
+    def __unicode__(self):
+        self.has_other_parameters
+
+    @classmethod
+    def create(cls, **kwargs):
+        if not 'has_crop_rotation' in kwargs:
+            raise ValidationError("SWATModelParameters has_crop_rotation is missing.")
+        else:
+            kwargs['has_crop_rotation'] = 1 if kwargs['has_crop_rotation'] == 'True' else 0
+        if not 'has_title_drainage' in kwargs:
+            raise ValidationError("SWATModelParameters has_title_drainage is missing.")
+        else:
+            kwargs['has_title_drainage'] = 1 if kwargs['has_title_drainage'] == 'True' else 0
+        if not 'has_point_source' in kwargs:
+            raise ValidationError("SWATModelParameters has_point_source is missing.")
+        else:
+            kwargs['has_point_source'] = 1 if kwargs['has_point_source'] == 'True' else 0
+        if not 'has_fertilizer' in kwargs:
+            raise ValidationError("SWATModelParameters has_fertilizer is missing.")
+        else:
+            kwargs['has_fertilizer'] = 1 if kwargs['has_fertilizer'] == 'True' else 0
+        if not 'has_tillage_operation' in kwargs:
+            raise ValidationError("SWATModelParameters has_tillage_operation is missing.")
+        else:
+            kwargs['has_tillage_operation'] = 1 if kwargs['has_tillage_operation'] == 'True' else 0
+        if not 'has_inlet_of_draining_watershed' in kwargs:
+            raise ValidationError("SWATModelParameters has_inlet_of_draining_watershed is missing.")
+        else:
+            kwargs['has_inlet_of_draining_watershed'] = 1 if kwargs['has_inlet_of_draining_watershed'] == 'True' else 0
+        if not 'has_irrigation_operation' in kwargs:
+            raise ValidationError("SWATModelParameters has_irrigation_operation is missing.")
+        else:
+            kwargs['has_irrigation_operation'] = 1 if kwargs['has_irrigation_operation'] == 'True' else 0
+        if not 'has_other_parameters' in kwargs:
+            raise ValidationError("SWATModelParameters has_other_parameters is missing.")
```

Comment:
A lot of these `create()` and `update()` methods are doing things model validation should be responsible for.